RIOT-OS/RIOT
# shell/Kconfig: make config options usable with dependency resolution

### Contribution description
Currently, one can use Kconfig to configure the shell, but not when using Kconfig for dependency resolution. This adds that as an option.



### Testing procedure
```sh
make -C tests/test_tools/ clean menuconfig
```
should look like it does in master.

```sh
TEST_KCONFIG=1 make -C tests/test_tools/ clean menuconfig
```
should now include the configuration options under System -> Shell interpreter

![image](https://user-images.githubusercontent.com/675644/108233214-d67a0300-7143-11eb-9362-ad6823df6c74.png)




### Issues/PRs references
Needed for a nicely integrated #15951 for dependency resolution using Kconfig.
